File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int start = 0;
        int middle = 0;
        int last = nums.size() -1;

        while(middle <= last){
            switch (nums[middle]){
                case 0:
                    swap(nums[start++], nums[middle++]);
                    break;
                case 1:
                    middle++;
                    break;
                case 2:
                    swap(nums[middle], nums[last--]);
                    break;
            }
        }
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int n = nums.length;
        int left = 0;
        int right = n-1;
        int pointer = 0;
        while(pointer<=right){
            if(nums[pointer] == 0){
                int temp = nums[pointer];
                nums[pointer] = nums[left];
                nums[left] = temp;
                left++;
                pointer++;
            }else if(nums[pointer] == 2){
                 int temp = nums[pointer];
                nums[pointer] = nums[right];
                nums[right] = temp;
                right--;
            }else{
               pointer++;
            }
        }

    }
}